pkg/sockets/tcp_socket.go

```diff
@@ -4,11 +4,56 @@ package sockets
 
 import (
 	"crypto/tls"
+	"fmt"
 	"net"
 	"net/http"
+	"net/url"
+	"os"
+	"strings"
 	"time"
+
+	"golang.org/x/net/proxy"
 )
 
+// The uppercase and the lowercase are available for the proxy settings.
+// See the Go specification for details on these variables. https://golang.org/pkg/net/http/
+func GetProxyEnv(key string) string {
```

Comment:
I know this probably isn't the ideal place, but both `daemon` and `sockets` module needed this function, so let me know if you have a preference on where it should go.